caproto/threading/client.py

```diff
@@ -434,17 +436,20 @@ def send(self, port, *commands):
         """
         bytes_to_send = self.broadcaster.send(*commands)
         for host in ca.get_address_list():
-            self.broadcaster.log.debug('sending bytes to %s', host)
             try:
                 if ':' in host:
-                    host, _, specified_port = host.partition(':')
-                    self.udp_sock.sendto(bytes_to_send, (host,
-                                                         int(specified_port)))
+                    host, _, port_as_str = host.partition(':')
```

Comment:
OK - now my recommendation changes from the last PR. Can you move these out of the `try/except OSError` block?